### PR TITLE
Lecture page

### PR DESCRIPTION
**Pull request by alexgirard**
- the reactions have a diagonal up animation when pressed
- added `@babel/plugin-proposal-class-properties` for compatibility with youtube video library

image

### CODE REVIEW

File: src/components/ReactionBar.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    border-right: 1px solid #d9dbe9;
+  }
+`;
+
+const emojis = ['😃', '🙁', '✋', '❓', '👍', '👎', '⏩', '⏪'];
+
+const ReactionBar = props => (
+  <ReactionContainer>
+    {emojis.map((emoji, index) => (
+      <Emoji key={`${index}-${emoji}`} emoji={emoji} {...props} />
```
Comment (BenLanglois) [inline src/components/ReactionBar.jsx:24]: The `onClick` callback prop should probably be called with the emoji type. `Emoji` calls it with no arguments, so it should be set here.

File: src/components/YoutubePlayer.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    position: absolute;
+    top: 0;
+    left: 0;
+  }
+`;
+
+class YoutubePlayer extends React.Component {
+  ref = player => (this.player = player);
+
+  getCurTime = () => this.player.getCurrentTime();
```
Comment (BenLanglois) [inline src/components/YoutubePlayer.jsx:20]: `this.player` could be undefined if the component hasn't rendered yet, so calling `getCurrentTime` could throw an exception.

File: src/components/YoutubePlayer.jsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+  > .react-player {
+    position: absolute;
+    top: 0;
+    left: 0;
+  }
+`;
+
+class YoutubePlayer extends React.Component {
+  ref = player => (this.player = player);
```
Comment (BenLanglois) [inline src/components/YoutubePlayer.jsx:18]: I believe callback refs are not the preferred way of doing refs anymore in react. Instead there's `React.createRef()` or the `useRef` hook.

File: src/pages/Lecture.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+const Title = styled.h1`
+  margin-bottom: 0.5em;
+`;
+
+const VideoContainer = styled.div`
+  border: 1px solid #d9dbe9;
+`;
+
+class Lecture extends React.Component {
+  getCurTime = () => this.child.getCurTime();
```
Comment (BenLanglois) [inline src/pages/Lecture.jsx:18]: Same thing here: `this.child` may be undefined if it has not been rendered yet.

**Review by BenLanglois — APPROVED** [review]

File: src/components/YoutubePlayer.jsx (hunk 3 above)
Comment (alexgirard) [reply src/components/YoutubePlayer.jsx:18]: HMM you are correct, I was just following along with some dependency documentation and didn't realize. Will merge for now but will make a note to fix later.

**Pull request merged**